include/api/CDataFrameAnalyzer.h

```diff
@@ -99,6 +101,7 @@ class API_EXPORT CDataFrameAnalyzer {
     TDataFrameUPtr m_DataFrame;
     TTemporaryDirectoryPtr m_DataFrameDirectory;
     TJsonOutputStreamWrapperUPtrSupplier m_ResultsStreamSupplier;
+    TRapidJsonConcurrentLineWriterPtr m_ResultsWriter;
```

Comment:
I don't think this can work: `CDataFrameAnalyzer` has automatic storage and is created at this [line](https://github.com/elastic/ml-cpp/blob/master/bin/data_frame_analyzer/Main.cc#L207). Handle fatal calls `std::exit` which states "[The] stack is not unwound: destructors of variables with automatic storage duration are not called", see [here](https://en.cppreference.com/w/cpp/utility/program/exit). Objects with static storage are destructed by calling `std::exit` so one possibility would be temporarily register a callback to call the destructor with with an object with static storage.